internal/controller/finbackupconfig_controller.go

```diff
@@ -52,5 +102,158 @@ func (r *FinBackupConfigReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 func (r *FinBackupConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&finv1.FinBackupConfig{}).
+		Owns(&batchv1.CronJob{}).
 		Complete(r)
 }
+
+func getCephClusterIDFromSCName(ctx context.Context, k8sClient client.Client, storageClassName string) (string, error) {
+	var storageClass storagev1.StorageClass
+	err := k8sClient.Get(ctx, types.NamespacedName{Name: storageClassName}, &storageClass)
+	if err != nil {
+		return "", fmt.Errorf("failed to get StorageClass: %s: %w", storageClassName, err)
+	}
+
+	if !strings.HasSuffix(storageClass.Provisioner, ".rbd.csi.ceph.com") {
+		return "", fmt.Errorf("SC is not managed by RBD: %s: %w", storageClassName, errEmptyClusterID)
+	}
+	clusterID, ok := storageClass.Parameters["clusterID"]
+	if !ok {
+		return "", fmt.Errorf("clusterID not found: %s: %w", storageClassName, errEmptyClusterID)
+	}
+
+	return clusterID, nil
+}
+
+func getCephClusterIDFromPVC(
+	ctx context.Context,
+	k8sClient client.Client,
+	pvc *corev1.PersistentVolumeClaim,
+) (string, error) {
+	logger := log.FromContext(ctx)
+
+	storageClassName := pvc.Spec.StorageClassName
+	if storageClassName == nil {
+		logger.Info("not managed storage class", "namespace", pvc.Namespace, "pvc", pvc.Name)
+		return "", nil
+	}
+
+	clusterID, err := getCephClusterIDFromSCName(ctx, k8sClient, *storageClassName)
+	if err != nil {
+		logger.Info("failed to get ceph cluster ID from StorageClass name",
+			"error", err, "namespace", pvc.Namespace, "pvc", pvc.Name, "storageClassName", *storageClassName)
+		if errors.Is(err, errEmptyClusterID) {
+			return "", nil
+		}
+		return "", err
+	}
+
+	return clusterID, nil
+}
+
+func getRunningPod(ctx context.Context, client client.Client) (*corev1.Pod, error) {
+	name, ok := os.LookupEnv("POD_NAME")
```

Comment:
Currently `POD_NAME` isn't defined in Deployment spec (cf. https://github.com/cybozu-go/fin/blob/358835532285e8c5244b870ace85b5215b359df7/config/manager/manager.yaml#L70-L80), so this spec should be updated.